src/Data/Nat/Properties.agda

```diff
@@ -2098,8 +2098,8 @@ _>″?_ = flip _<″?_
 
 ≤‴⇒≤″ : ∀{m n} → m ≤‴ n → m ≤″ n
 ≤‴⇒≤″ {m = m} ≤‴-refl     = less-than-or-equal {k = 0} (+-identityʳ m)
-≤‴⇒≤″ {m = m} (≤‴-step x) = less-than-or-equal (trans (+-suc m _) (_≤″_.proof ind)) where
-  ind = ≤‴⇒≤″ x
+≤‴⇒≤″ {m = m} (≤‴-step x) with less-than-or-equal proof ← ≤‴⇒≤″ x
+  = less-than-or-equal (trans (+-suc m _) proof)
```

Comment:
can't you use proj_2 instead of `with` here?

Reply:
Ah!... nice question: if you do use `proj₂`, then you are exposing the (new) representation as a Σ-type; whereas, by pattern matching on the synonym, you preserve the original abstraction. This also arises (I realised later) in the very definition of the function `proof` that I introduce for the sake of having a name to deprecate...